src/tiled/objectselectiontool.cpp

```diff
@@ -533,6 +535,43 @@ void ObjectSelectionTool::mousePressed(QGraphicsSceneMouseEvent *event)
 
         break;
     }
+    case Qt::RightButton:
+        if (event->modifiers() & Qt::AltModifier) {
+            QList<MapObjectItem*> underlyingObjects = objectItemsAt(event->scenePos());
+            if (underlyingObjects.empty())
+                break;
+            QMenu selectUnderlyingMenu;
+
+            for (int levelNum = 0; levelNum < underlyingObjects.size(); ++levelNum) {
+                const QString& objectName = underlyingObjects[levelNum]->mapObject()->name();
+                QString actionName = (objectName.isEmpty() ? tr("Object at level %n", "", levelNum + 1) : objectName)
+                        + tr(levelNum ? "" : " (topmost)");
```

Comment:
Sure, that would be fine. Then you can even do `"&1) "` for the first 9 entries to enable the number keys to be pressed for selecting one of the entries (`&` sets the mnemonic).